FileManager/CommandHandlers/DirectoryCommandHandler.cs

```diff
@@ -0,0 +1,407 @@
+﻿using FileManager.CommandHandlers.Base;
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace FileManager.CommandHandlers;
+public class DirectoryCommandHandler : CommandHandler
+{
+    private readonly IUserInterface _UserInterface;
+    private readonly DirectoryInfo _Directory;
+
+    private void CopyCatalogue(DirectoryInfo dir, string destinationDir)
+    {
+        DirectoryInfo[] dirs = dir.GetDirectories();
+
+        // Создаёт текущий каталог в папке назначения
+        Directory.CreateDirectory(destinationDir);
+
+        // Файлы в исходном каталоге копируются в конечный
+        foreach (FileInfo file in dir.GetFiles())
```

Comment:
```suggestion
        foreach (FileInfo file in dir.EnumerateFiles())
```
Метод не выделяет память под хранение всех файлов (а вдруг их в каталоге 1 000 000? Пока система их все не прочитает и не создаст массив, метод `GetFiles` не отпустит), а обрабатывает файлы по одному за итерацию.